src/Development/Shake/Command.hs

```diff
@@ -219,19 +220,31 @@ parseFSAT = mapMaybe (f . wordsBy (== ':')) . lines
           f _ = Nothing
 
 
-unixWhitelist :: IO [FilePath]
-unixWhitelist = do
+unixWhitelist :: [FilePath]
+unixWhitelist = unsafePerformIO $ do
     home <- getEnv "HOME"
-    return [home ++ "/.ghc"
-           ,home ++ "/Library/Haskell"
-           ,home ++ "/Applications"
-           ,home ++ "/.cabal"
-           ,"/Applications"
-           ,"/var/folders"
-           ,"/usr"
-           ,"/Library"
-           ,"/System"
-           ]
+    ghcPath <- lookupEnv "GHC_PACKAGE_PATH"
+    sbcPath <- lookupEnv "CABAL_SANDBOX_CONFIG"
+    cabPath <- lookupEnv "CABAL_SANDBOX_PACKAGE_PATH"
+    let sep = if isWindows then ';' else ':'
+        splitted = map dropFileName . wordsBy (== sep) . fromMaybe ""
```

Comment:
It's better to use `splitSearchPath` from the filepath library which does the splitting and always does the right thing on Windows vs Linux.